common/config_file.h

```diff
@@ -6,9 +6,9 @@
 // Reads an config file, looking for variable assignments.
 // TODO(hubbe): Read config files from serialflash.
 struct ConfigFile {
-  void Read(FileReader* f) {
+  bool Read(FileReader* f) {
     SetVariable("=", 0.0);  // This resets all variables.
-    if (!f || !f->IsOpen()) return;
+    if (!f || !f->IsOpen()) return false;
```

Comment:
Good start, but not enough.
We need to know if the file is truncated or not, so I think we need to also check that there is an "end" at the end of the file.

Might be a good idea to return an enum with three values, something like:

CONFIG_FILE_READ_FAIL
CONFIG_FILE_READ_OK
CONFIG_FILE_READ_OK_WITH_END

Since hand-written config files aren't expected to have "end" at the end, only automatically generated files will have that.

Reply:
It occured to me at lunch that there is a much easier way to do this.
We could make "end=1" be the end marker, then add an "end" variable to SaveStateFile, and default it to zero.
Then savefile.end will tell us if the file was read successfully or not.